lang/src/main/test/java/org/apache/shiro/util/ClassUtilsTest.java

```diff
@@ -0,0 +1,89 @@
+        Assert.assertEquals(ClassUtils.forName("double"), double.class);
+        Assert.assertEquals(ClassUtils.forName("void"), void.class);
+
+        Assert.assertEquals(ClassUtils.forName(boolean.class.getName()), boolean.class);
+        Assert.assertEquals(ClassUtils.forName(byte.class.getName()), byte.class);
+        Assert.assertEquals(ClassUtils.forName(char.class.getName()), char.class);
+        Assert.assertEquals(ClassUtils.forName(short.class.getName()), short.class);
+        Assert.assertEquals(ClassUtils.forName(int.class.getName()), int.class);
+        Assert.assertEquals(ClassUtils.forName(long.class.getName()), long.class);
+        Assert.assertEquals(ClassUtils.forName(float.class.getName()), float.class);
+        Assert.assertEquals(ClassUtils.forName(double.class.getName()), double.class);
+        Assert.assertEquals(ClassUtils.forName(void.class.getName()), void.class);
+
+    }
+
+    @Test
+    public void testGetPrimitiveArrays() throws UnknownClassException {
+
+        Assert.assertEquals(ClassUtils.forName("[Z"), boolean[].class);
+        Assert.assertEquals(ClassUtils.forName("[B"), byte[].class);
+        Assert.assertEquals(ClassUtils.forName("[C"), char[].class);
+        Assert.assertEquals(ClassUtils.forName("[S"), short[].class);
+        Assert.assertEquals(ClassUtils.forName("[I"), int[].class);
+        Assert.assertEquals(ClassUtils.forName("[J"), long[].class);
+        Assert.assertEquals(ClassUtils.forName("[F"), float[].class);
+        Assert.assertEquals(ClassUtils.forName("[D"), double[].class);
+
+
+        Assert.assertEquals(ClassUtils.forName(boolean[].class.getName()), boolean[].class);
+        Assert.assertEquals(ClassUtils.forName(byte[].class.getName()), byte[].class);
+        Assert.assertEquals(ClassUtils.forName(char[].class.getName()), char[].class);
+        Assert.assertEquals(ClassUtils.forName(short[].class.getName()), short[].class);
+        Assert.assertEquals(ClassUtils.forName(int[].class.getName()), int[].class);
+        Assert.assertEquals(ClassUtils.forName(long[].class.getName()), long[].class);
+        Assert.assertEquals(ClassUtils.forName(float[].class.getName()), float[].class);
+        Assert.assertEquals(ClassUtils.forName(double[].class.getName()), double[].class);
+    }
+
+    @Test
+    public void testGetClass() {
+        Assert.assertEquals(ClassUtils.forName("java.lang.String"), String.class);
+        Assert.assertEquals(ClassUtils.forName("[Ljava.lang.String;"), String[].class);
+        Assert.assertEquals(ClassUtils.forName(String.class.getName()), String.class);
+        Assert.assertEquals(ClassUtils.forName(String[].class.getName()), String[].class);
+
+        Assert.assertEquals(ClassUtils.forName("org.apache.shiro.util.ClassUtilsTest"), ClassUtilsTest.class);
+        Assert.assertEquals(ClassUtils.forName("[Lorg.apache.shiro.util.ClassUtilsTest;"), ClassUtilsTest[].class);
```

Comment:
When put into the correct path and package, this test will fail:
`org.apache.shiro.lang.util.UnknownClassException: Unable to load class named [[Lorg.apache.shiro.util.ClassUtilsTest;] from the thread context, current, or system/application ClassLoaders.  All heuristics have been exhausted.  Class could not be found.`